apps/src/applab/commands.js

```diff
@@ -866,6 +874,13 @@ applabCommands.setImageURL = function (opts) {
   var element = document.getElementById(opts.elementId);
   if (divApplab.contains(element) && element.tagName === 'IMG') {
     element.src = Applab.maybeAddAssetPathPrefix(opts.src);
+
+    if (!ensureCached[element.src]) {
+      ensureCached[element.src] = true;
+      var img = new Image();
+      img.src = element.src;
```

Comment:
In this case "cached" is a little confusing.  It just means we've initiated an open-ended request for the asset.  The asset may still be loading.

The problem was that quickly-changing images would keep getting canceled and never make it into the cache.  This ensures that even if `element.src` is changed (which would normally cancel the download), `img.src` will be unchanged and ensure the asset is fully downloaded into the browser's cache.